### PR TITLE
refactor: use npm-registry-fetch

### PR DESCRIPTION
**Pull request by zkat**
This is the refactor to make `npm-profile` use the new registry fetch library, which makes it so we can consistently handle all the various incidentals of networking stuff and config and all that jazz without random adapters in npm core.

This is a **BREAKING CHANGE** because it's changing the options behavior. I've updated all the docs to reflect this, since most of those configs will now go through `n-r-f`.

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -16,96 +18,79 @@ exports.listTokens = listTokens
...
 exports.createToken = createToken
 
+const ProfileConfig = pudding({
+  creds: {},
+  hostname: {}
+})
+
 // try loginWeb, catch the "not supported" message and fall back to couch
-function login (opener, prompter, conf) {
+function login (opener, prompter, opts) {
```
Comment (iarna) [inline lib/index.js]: The conf/opts rename makes this diff a lot noiser and harder to read than it would otherwise be. It's the sort of thing that for that reason I think should be a separate commit.

File: lib/index.js [hunk 2]
```diff
@@ -156,66 +166,67 @@ function loginCouch (username, password, conf) {
...
-        auth: {
-          basic: {
-            username: username,
-            password: password
-          }
-        }
-      }
-      return fetchJSON(Object.assign({}, conf, req))
+        body,
+        username,
```
Comment (iarna) [inline lib/index.js]: basic auth is the default, I take it?

File: lib/index.js [hunk 3]
```diff
@@ -226,27 +237,27 @@ function listTokens (conf) {
...
+function removeToken (tokenKey, opts) {
   validate('SO', arguments)
-  const target = url.resolve(conf.registry, `-/npm/v1/tokens/token/${tokenKey}`)
-  return fetchJSON(Object.assign({target: target, method: 'DELETE'}, conf))
+  const target = `/-/npm/v1/tokens/token/${tokenKey}`
+  return fetch(target, ProfileConfig(opts, {
+    method: 'DELETE'
+  })).then(res => {
+    res.body.resume()
+    return null
```
Comment (iarna) [inline lib/index.js]: This whole thing does feel awkward, but I suppose not caring about a response body is unusual enough that it's fine.

File: lib/index.js [hunk 4]
```diff
@@ -256,24 +267,12 @@ class HttpErrorBase extends Error {
     this.statusCode = res.status
     this.code = 'E' + res.status
     this.method = method
-    this.target = target
+    this.target = res.url
```
Comment (iarna) [inline lib/index.js]: how does `res.url` differ from `target` and if they don't, why do we still have `target`?

File: lib/index.js [hunk 5]
```diff
@@ -291,104 +290,18 @@ class WebLoginNotSupported extends HttpErrorBase {
...
-    } else {
-      return retVal
-    }
-  })
-}
-
 function packageName (href) {
   try {
     let basePath = url.parse(href).pathname.substr(1)
     if (!basePath.match(/^-/)) {
```
Comment (iarna) [inline lib/index.js]: This whole if can certainly go away, as all URLs requested in this library start with `-`. This maybe should be replicated over in npm-registry-fetch, however?

This function is used to set the `pkgid` attribute on errors, which is in turn used by npm to provide better error messages with more context to users.

File: lib/index.js (hunk 2 above)
Comment (zkat) [reply lib/index.js]: ooh. No it's not. Good catch 😱

File: lib/index.js (hunk 4 above)
Comment (zkat) [reply lib/index.js]: I don't know, tbh. I think we can probably drop this.

File: lib/index.js (hunk 1 above)
Comment (zkat) [reply lib/index.js]: I've moved it into a separate commit

File: lib/index.js (hunk 3 above)
Comment (zkat) [reply lib/index.js]: I've added `opts.ignoreBody` to `n-r-f`, which I think will help this pattern suck a bit less in the future. I'd rather just do that than do some weird heuristics, tbh. There's already `DELETE` API calls, for example, that _return the old value in the body_, so.
